src/abi_impl.rs

```diff
@@ -163,14 +163,18 @@ pub(crate) fn assembly_script_call_module(
     let memory = get_memory!(env);
     let address = &get_string(memory, address)?;
     let function = &get_string(memory, function)?;
-    let param = &get_string(memory, param)?;
-    let response = call_module(env, address, function, param, call_coins)?;
-    match StringPtr::alloc(&response.ret, &env.wasm_env) {
-        Ok(ret) => Ok(ret.offset() as i32),
-        _ => abi_bail!(format!(
-            "Cannot allocate response in call {}::{}",
-            address, function
-        )),
+
+    let param = get_exported_ptr(memory, param)?;
+    let response = call_module(env, address, function, &Some(param), call_coins)?;
+    match response.ret {
+        Some(ret) => match AnyPtr::import(&ret, &env.wasm_env) {
+            Ok(ret_ptr) => Ok(ret_ptr.offset() as i32),
+            Err(err) => abi_bail!(format!(
+                "cannot allocate response in call {}::{}\n{}",
+                address, function, err
+            )),
+        },
+        None => abi_bail!("calling a module should return a value"),
```

Comment:
If no return value, I choose to return an error but I can also return an empty string

Reply:
Is void considered as "no return value"? If yes it shouldn't be an error IMO.